src/main/java/com/vaadin/demo/component/treegrid/TreeGridRichContent.java

```diff
@@ -0,0 +1,107 @@
+package com.vaadin.demo.component.treegrid;
+
+import com.vaadin.demo.DemoExporter;
+import com.vaadin.demo.domain.DataService;
+import com.vaadin.demo.domain.Person;
+import com.vaadin.flow.component.avatar.Avatar;
+import com.vaadin.flow.component.html.Anchor;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.orderedlayout.FlexComponent;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.treegrid.TreeGrid;
+import com.vaadin.flow.router.Route;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Route("tree-grid-rich-content")
+public class TreeGridRichContent extends Div {
+
+    private List<Person> people = DataService.getPeople();
+
+    public TreeGridRichContent() {
+        TreeGrid<Person> treeGrid = new TreeGrid<>();
+        treeGrid.setItems(getManagers(), this::getStaff);
+
+        // tag::snippet[]
+        treeGrid.addComponentHierarchyColumn(person -> {
+            Avatar avatar = new Avatar();
+            avatar.setName(person.getFullName());
+            avatar.setImage(person.getPictureUrl());
+
+            Span fullName = new Span(person.getFullName());
+
+            Span profession = new Span(person.getProfession());
+            profession.getStyle()
+                    .set("color", "var(--lumo-secondary-text-color)")
+                    .set("font-size", "var(--lumo-font-size-s)");;
+
+            VerticalLayout column = new VerticalLayout(fullName, profession);
+            column.getStyle()
+                    .set("line-height", "var(--lumo-line-height-m)");
+            column.setPadding(false);
+            column.setSpacing(false);
+
+            HorizontalLayout row = new HorizontalLayout(avatar, column);
+            row.setAlignItems(FlexComponent.Alignment.CENTER);
+            row.setSpacing(true);
+            return row;
+        });
```

Comment:
The header is missing:
```suggestion
        }).setHeader("Employee");
```